retrofit/src/main/java/retrofit2/ParameterHandler.java

```diff
@@ -176,34 +197,46 @@
         builder.addQueryParam(entryKey, convertedEntryValue, encoded);
       }
     }
+
+    private void throwParameterError(String message) {
```

Comment:
These helper methods should either be inlined or you should move it to the abstract class and pass the `Method` and parameter index up